ocfweb/lab_reservations/reserve.py

```diff
@@ -0,0 +1,168 @@
+from collections import namedtuple
+from datetime import datetime
+
+import ocflib.misc.validators
+from django import forms
+from django.shortcuts import redirect
+from django.shortcuts import render
+from django.urls import reverse
+from jinja2 import Environment
+from jinja2 import PackageLoader
+from ocflib.account.search import user_attrs_ucb
+from ocflib.misc.mail import send_mail
+
+from ocfweb.auth import calnet_required
+from ocfweb.component.forms import wrap_validator
+
+JINJA_MAIL_ENV = Environment(loader=PackageLoader('ocfweb', ''))
+
+
+class NewReservationRequest(namedtuple(
+    'NewReservationRequest', [
+        'real_name',
+        'contact_email',
+        'group',
+        'reason',
+        'date',
+        'starttime',
+        'endtime',
+    ],
+)):
+
+    def to_dict(self):
+        return self._asdict()
```

Comment:
a little nitpicky, but i'd just have consumers call `_asdict()` as this is a standard method (i don't need to look at the class implementation to see what it does, i'll know it does the same thing as `_asdict` on every other namedtuple)

this is all a little muddied by the fact that it's an underscored method (which usually means "private", but we frequently ignore in namedtuples)

ok to keep it either way, up to you